starlite/response/base.py

```diff
@@ -0,0 +1,328 @@
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Generic,
+    List,
+    Optional,
+    Tuple,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from orjson import OPT_INDENT_2, OPT_OMIT_MICROSECONDS, OPT_SERIALIZE_NUMPY, dumps
+from pydantic_openapi_schema.v3_1_0 import OpenAPI
+from yaml import dump as dump_yaml
+
+from starlite.datastructures import Cookie
+from starlite.enums import MediaType, OpenAPIMediaType
+from starlite.exceptions import ImproperlyConfiguredException
+from starlite.status_codes import (
+    HTTP_200_OK,
+    HTTP_204_NO_CONTENT,
+    HTTP_304_NOT_MODIFIED,
+)
+from starlite.utils.helpers import get_enum_string_value
+from starlite.utils.serialization import default_serializer
+
+if TYPE_CHECKING:
+    from typing_extensions import Literal
+
+    from starlite.datastructures import BackgroundTask, BackgroundTasks
+    from starlite.types import (
+        HTTPResponseBodyEvent,
+        HTTPResponseStartEvent,
+        Receive,
+        ResponseCookies,
+        Scope,
+        Send,
+    )
+
+T = TypeVar("T")
+
+
+class Response(Generic[T]):
+    __slots__ = (
+        "status_code",
+        "media_type",
+        "background",
+        "headers",
+        "cookies",
+        "encoding",
+        "body",
+        "status_allows_body",
+        "is_head_response",
+    )
+
+    def __init__(
+        self,
+        content: T,
+        *,
+        status_code: int = HTTP_200_OK,
+        media_type: Union[MediaType, "OpenAPIMediaType", str] = MediaType.JSON,
+        background: Optional[Union["BackgroundTask", "BackgroundTasks"]] = None,
+        headers: Optional[Dict[str, Any]] = None,
+        cookies: Optional["ResponseCookies"] = None,
+        encoding: str = "utf-8",
+        is_head_response: bool = False,
+    ) -> None:
+        """This is the base Starlite HTTP response class, its is used as a
```

Comment:
Typo: "its is used" > "it is used"
Typo: "Used as a basis" > "Used as the basis"

Reply:
fixed